3p/iframe-transport-client.js

```diff
@@ -33,61 +33,131 @@ export class IframeTransportClient {
     /** @private {!Window} */
     this.win_ = win;
 
-    /** @private {?function(string,string)} */
-    this.listener_ = null;
+    /** @private {Object<string,IframeTransportContext>} */
+    this.creativeIdToContext_ = {};
+
+    const parsedFrameName = tryParseJson(this.win_.name);
+
+    /** @private {string} */
+    this.vendor_ = dev().assertString(parsedFrameName['type'],
+        'Parent frame must supply vendor name as type in ' +
+        this.win_.location.href);
+    dev().assert(this.vendor_.length > 0, 'Vendor name cannot be empty in ' +
```

Comment:
Don't need >.  We don't actually verify that this vendor matches one that allows 3p and I think that is fine (limit growth of the lib) but can you add a comment on what would happen if it were a vendor that was not in the list?